classes/persistent/sitenotice.php

```diff
@@ -185,13 +144,7 @@ public static function reset($noticeid) {
     public static function get_enabled_notices(): array {
         if (!$result = self::get_enabled_notices_cache()->get('records')) {
             $select = "enabled = ? AND ((timeend = 0 AND timestart = 0) OR (timeend <> 0 AND timestart <> 0 AND ? < timeend))";
-            $persistents = self::get_records_select($select, [1, time()], 'id');
-            $result = [];
-            foreach ($persistents as $persistent) {
-                $record = $persistent->to_record();
-                $result[$record->id] = $record;
-            }
-
+            $result = self::get_records_select($select, [1, time()], 'id');
             self::get_enabled_notices_cache()->set('records', $result);
```

Comment:
Do we need to trigger upgrade to rebuild this cache? Otherwise we can have mix of records?

Reply:
Error I got before purging caches.

![image](https://user-images.githubusercontent.com/4626431/175460591-92995ddc-9626-438b-8833-a7a5e972c8e6.png)

Reply:
Yup, added version bump.